appcheck/firebase-appcheck-recaptchaenterprise/src/main/java/com/google/firebase/appcheck/recaptchaenterprise/internal/ProviderMultiResourceComponent.java

```diff
@@ -0,0 +1,47 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.appcheck.recaptchaenterprise.internal;
+
+import androidx.annotation.NonNull;
+import dagger.assisted.Assisted;
+import dagger.assisted.AssistedFactory;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+/** Multi-resource container for RecaptchaEnterpriseAppCheckProvider */
+@Singleton
+public final class ProviderMultiResourceComponent {
+  private final RecaptchaEnterpriseAppCheckProviderFactory providerFactory;
+
+  private final Map<String, RecaptchaEnterpriseAppCheckProvider> instances =
+      new ConcurrentHashMap<>();
+
+  @Inject
+  ProviderMultiResourceComponent(RecaptchaEnterpriseAppCheckProviderFactory providerFactory) {
+    this.providerFactory = providerFactory;
+  }
+
+  @NonNull
+  public RecaptchaEnterpriseAppCheckProvider get(@NonNull String siteKey) {
+    return instances.computeIfAbsent(siteKey, providerFactory::create);
```

Comment:
`computeIfAbsent` is API 24 

https://developer.android.com/reference/java/util/Map#computeIfAbsent(K,%20java.util.function.Function%3C?%20super%20K,?%20extends%20V%3E)